src/agent/onefuzz-agent/src/tasks/config.rs

```diff
@@ -79,9 +81,16 @@ pub enum Config {
 }
 
 impl Config {
-    pub fn from_file(path: impl AsRef<Path>) -> Result<Self> {
-        let json = std::fs::read_to_string(path)?;
-        Ok(serde_json::from_str(&json)?)
+    pub fn from_file(path: impl AsRef<Path>, setup_dir: Option<impl AsRef<Path>>) -> Result<Self> {
+        let file = std::fs::File::open(path)?;
+        let reader = std::io::BufReader::new(file);
```

Comment:
I think it's both simpler and probably more efficient to just use `fs::read_to_string()` or `fs::read()` here. Those functions are meant for one bulk read from a file, pre-allocate a buffer based on file size, &c. Was there a specific reason you changed this?